Feature/viper config

Faheetah opened the pull request:
Doing some maintenance mostly. Highlights:

Caught up lint/vet even if the //ObjName comments are crummy
Added a noop auth as a default instead of LDAP, so it's easier to just pick up and try, and not dependent on any configuration
RethinkDB can be skipped on tests if needed, handy if you want to work a module that doesn't need it but can't spin it up off hand (would be good for travis jobs)
Fixed CORS so we can deploy the UI separately, this is the main change so I can rewrite the UI without having to bother with directory shares, just directly dev locally through yarn start
Reworked the docker image to just be from scratch, used a docker pipeline to build it with the _/golang image, huge win here switched to viper so we don't need near as much logic for env vars, and was able to remove dockerize without consequence as a result

I'd like to propose ditching the vendor/ folder, I added it on the .gitignore here because it was crazy trying to get vscode to even comprehend how many files changed just removing burntsushi/toml and adding viper. I think we can just glide up when we clone. The deps are pinned to commits in the lock so there's nowhere near enough risk that they will get out of sync either.

Faheetah committed:
Use multi-stage docker build to scratch container

Faheetah committed:
Conform to lint/vet

Faheetah committed:
Add noop auth as default auth

Faheetah committed:
Refactor tests to allow skipping rethinkdb or use env var for address

Faheetah committed:
Allow CORS for external UIs

Faheetah committed:
Use Viper and natively support environment variables

Faheetah committed:
Stop tracking vendor/ it's becoming unwield

allen13 commented:
Having the vendor directory committed really helps with build stability. It's a little bit messy but avoids problems like the infamous nodejs pad left dependency that broke the internet. What I usually do is separate dependency changes into separate commits. Update dependencies -> glide update -> git add . -> git commit -m "updating depencies" 

https://www.theregister.co.uk/2016/03/23/npm_left_pad_chaos/

Faheetah commented:
Yeah I remember the pad left nonsense. I was planning on doing a separate commit for them if we did keep it. Figured I would mention it since it does make IDEs a lot clunkier (just removing toml and adding viper was 800 file +-) with both search and git updates. I can add them in another commit and un-ignore it. If everything else looks good I'll run another commit this weekend for updated vendor.

Faheetah committed:
Fix middleware regression

Faheetah committed:
Default query for alert counts to empty map

Faheetah committed:
Clean up UI a bit

Faheetah commented:
Accidentally broked the jwt logic. Reverted it to restore ability to use multiple auth types (I see we use header and query string). Cleaned up a bit more, fixed the 500 from an empty /alerts/count and easier to click rows in the UI. If everything looks good I'll push up vendor.

Faheetah committed:
Return empty map for alert count on empty set

Faheetah committed:
Add vendor
